### ISSUE TITLE
add option to process folders/files whose name begin with a dot

### ISSUE THREAD

**Issue by daverusso** [eslint/eslint#10597]
**The version of ESLint you are using.**
5.1.0

**The problem you want to solve.**
I'm trying to integrate eslint into a product with "hidden" folders (that have named that begin with '.') containing javascript files.  I'm not able to change the names of the folder and, as far as I can tell, there's no option that allows eslint to process files in hidden folders.

**Your take on the correct solution to problem.**
Looking at the 5.1.0 code base, I see that ignore-paths.js _already has_ a option to disable the processing of dot files and folders.  The only thing missing is the addition of a command line option to set dotfiles to true.

I was able to fix my problem by adding
 ```
{ option: "dotfiles",
  type: "Boolean",
  default: "false",
  description: "Disable ignore of directories with a leading '.'"
},
```
at line 114 of eslint/lib/options.js, and
```
dotfiles: cliOptions.dotfiles,
```
at line 56 of eslint/lib/cli.js 

This issue is would also resolve  #10089

**Comment by platinumazure** [comment]
Just to make sure, are you able to unignore the dot folders/files in .eslintignore? We do prevent processing of dot folders/files by default, but it should be possible to unignore those.

.eslintignore:

```
!.someDotFolder/**/*.J's
```

CLI:

```bash
eslint src/ tests/ .someDotFolder/
```

**Comment by daverusso** [comment]
Aha!  That looks promising but I'm struggling with creating a .eslintignore that does my bidding.  I think it may be related to the fact that I'm passing absolute paths to eslint?

Also, I'd prefer to not have to place .eslintignore in the directories that I'm linting.  But, if I can get it to work, it's a much better workaround than me hacking the eslint sources.

```
$ ls -d /db/vendors/someproduct/source/ti/drivers/.meta
/db/vendors/someproduct/source/ti/drivers/.meta

$ cat /db/vendors/someproduct/source/ti/drivers/.eslintignore
!.meta/**/*.js

$ eslint /db/vendors/someproduct/source/ti/**/*.js

Oops! Something went wrong! :(

ESLint: 5.1.0.
All of the files matching the glob pattern "/db/vendors/someproduct/source/ti/**/*.js" are ignored.

If you don't want to lint these files, remove the pattern "/db/vendors/someproduct/source/ti/**/*.js" from the list of arguments passed to ESLint.
   :
```

**Comment by platinumazure** [comment]
You can put the .eslintignore in your project root. Ignore patterns are then processed relative to the location of the ignore file.

For example, if your project root is `/home/daverusso/projects/myProject` and you want to lint `/home/daverusso/projects/myProject/path/to/.something`:
* Create `/home/daverusso/projects/myProject/.eslintignore`
* Add this pattern: `!path/to/.something/**/*.js`
  * Not sure if you could just put the folder name with a trailing slash, try that too

ESLint only reads one .eslintignore file, in the current directory by default (or use `--ignore-path` CLI flag if you need to point elsewhere).

Hope this helps!

**Comment by daverusso** [comment]
Thanks (again)!  

Using the --ignore-path CLI flag I was able to get eslint to process files my "hidden" .meta directories.  It took some experimentation to figure out the right pattern to allow files under any .meta directory, but the following "ignore" file did the trick:
```
!.meta/
```
This file together with --ignore-path allows me to run eslint on any repo from anywhere and ensure the files in any .meta folder are processed.  Sweet.

As for this change request, there may be value in making it easier to turn off the hiding of dot files.  Using the git ignore syntax is certainly the right approach, but (because of it's completeness) it can be challenge to find the right pattern(s).  

Anyway, great tool and I love it's extensive configurability.
